apps/web/src/hooks/useFarcasterUserByFID.ts

```diff
@@ -0,0 +1,65 @@
+import { useQuery } from '@tanstack/react-query';
+
+// Type for the User profile
+type Bio = {
+  text: string;
+};
+
+type Profile = {
+  bio: Bio;
+};
+
+// Type for viewer context (the user's interaction with the viewer)
+type ViewerContext = {
+  following: boolean;
+  followed_by: boolean;
+};
+
+// Type for verified addresses
+type VerifiedAddresses = {
+  eth_addresses: string[];
+  sol_addresses: string[];
+};
+
+// Type for the main User object
+export type User = {
+  object: string;
+  fid: number;
+  custody_address: string;
+  username: string;
+  display_name: string;
+  pfp_url: string;
+  profile: Profile;
+  follower_count: number;
+  following_count: number;
+  verifications: string[];
+  verified_addresses: VerifiedAddresses;
+  active_status: string;
+  power_badge: boolean;
+  viewer_context: ViewerContext;
+};
+
+export type FIDResponse = {
+  users: User[];
+};
+
+const fetchFIDData = async (fid?: number | undefined): Promise<FIDResponse | undefined> => {
+  if (!fid) return;
+  const response = await fetch(`/farcaster/user?fid=${fid}`);
```

Comment:
Should we catch network errors?

Reply:
I think this is handled with !response.ok, and then secondarily by useQuery

lmk if I'm wrong and you'd prefer something else here